### PR TITLE
Allow placeholders in OpenAPI endpoints #2938

### LINKED ISSUE #2938: Allow OpenAPI planner to respect URLs with placeholders
In OpenAPI documentation, an endpoint might include a placeholder for a parameter:

```
GET /states/{abbr}
```

Currently, exact matches are needed with OpenAPI Planner to retrieve documentation. In the example above, `GET /states/FL` would receive a `ValueError(f"{endpoint_name} endpoint does not exist.")`.

### PR DESCRIPTION
**Pull request by jsoma**
Use regex matches when checking endpoints instead of exact matches. `{varname}` becomes `.*`

Fixes #2938

### CODE REVIEW

File: langchain/agents/agent_toolkits/openapi/planner.py [hunk 1]
```diff
@@ -218,12 +218,17 @@ def _create_and_run_api_controller_agent(plan_str: str) -> str:
             for method, route in matches
         ]
         endpoint_docs_by_name = {name: docs for name, _, docs in api_spec.endpoints}
```
Comment (dev2049) [inline langchain/agents/agent_toolkits/openapi/planner.py:220]: can we delete this

File: langchain/agents/agent_toolkits/openapi/planner.py [hunk 2]
```diff
@@ -218,12 +218,17 @@ def _create_and_run_api_controller_agent(plan_str: str) -> str:
...
+
         docs_str = ""
```
Comment (dev2049) [inline langchain/agents/agent_toolkits/openapi/planner.py:222]: could we factor our docs_str creation into a helper and write one or two simple unit tests for it?

**Comment by leo-gan** [comment]
@jsoma Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that ping me and I push this PR for the review. Thanks!

**Comment by codebanesr** [comment]
This issue still exists. Is someone looking into it ?

**Comment by baskaryan** [comment]
sorry for delay, thank you @jsoma!

**Pull request merged**